cmd/agent/daemon/metrics/agent.go

```diff
@@ -79,6 +83,21 @@ var (
 		Help: "Dropped exporter events",
 	}, []string{ExporterName})
 
+	AgentEnricherEventsTotalEnriched = promauto.NewCounterVec(prometheus.CounterOpts{
```

Comment:
Same comment as above, IMO we should avoid any such high cardinality metrics. This directly impacts customer's prometheus instances.